From cce24216e20eb09230d63069e3858a2169093f79 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Wed, 20 Feb 2019 11:34:14 +0100
Subject: [PATCH 1020/1051] arm64: dts: marvell: 91xx: complement CP0 SD
 description

Add missing V_3.3_SD regulator and card-detect description.

Change-Id: I557f97bc0c08ae90b923befca7fa1313b23da3a7
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/4415
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9130-db.dtsi | 17 +++++++++++++++--
 arch/arm64/boot/dts/marvell/cn9130.dtsi    |  4 ++++
 2 files changed, 19 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/cn9130-db.dtsi b/arch/arm64/boot/dts/marvell/cn9130-db.dtsi
index 82e813aa94a3..8bb4d8b8982e 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130-db.dtsi
@@ -41,7 +41,7 @@
 
 	cp0_reg_sd_vccq: cp0_sd_vccq@0 {
 		compatible = "regulator-gpio";
-		regulator-name = "cp0_sd_vcc";
+		regulator-name = "cp0_sd_vccq";
 		regulator-min-microvolt = <1800000>;
 		regulator-max-microvolt = <3300000>;
 		gpios = <&expander0 15 GPIO_ACTIVE_HIGH>;
@@ -49,6 +49,16 @@
 			  3300000 0x0>;
 	};
 
+	cp0_reg_sd_vcc: cp0_sd_vcc@0 {
+		compatible = "regulator-fixed";
+		regulator-name = "cp0_sd_vcc";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&expander0 14 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		regulator-always-on;
+	};
+
 	cp0_sfp_eth0: sfp-eth0 {
 		compatible = "sff,sfp";
 		i2c-bus = <&cp0_sfpp0_i2c>;
@@ -111,9 +121,12 @@
 
 &cp0_sdhci0 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&cp0_sdhci_pins>;
+	pinctrl-0 = <&cp0_sdhci_pins
+		     &cp0_sdhci_cd_pins>;
 	bus-width = <4>;
+	cd-gpios = <&cp0_gpio2 11 GPIO_ACTIVE_LOW>;
 	vqmmc-supply = <&cp0_reg_sd_vccq>;
+	vmmc-supply = <&cp0_reg_sd_vcc>;
 };
 
 &cp0_usb3_0 {
diff --git a/arch/arm64/boot/dts/marvell/cn9130.dtsi b/arch/arm64/boot/dts/marvell/cn9130.dtsi
index cff668a35d60..3a5f1ec8c68d 100644
--- a/arch/arm64/boot/dts/marvell/cn9130.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130.dtsi
@@ -86,6 +86,10 @@
 				       "mpp53", "mpp54", "mpp55";
 			marvell,function = "ge1";
 		};
+		cp0_sdhci_cd_pins: cp0-sdhci-cd-pins-0 {
+			marvell,pins = "mpp43";
+			marvell,function = "gpio";
+		};
 		cp0_sdhci_pins: cp0-sdhi-pins-0 {
 			marvell,pins = "mpp56", "mpp57", "mpp58",
 				       "mpp59", "mpp60", "mpp61";
-- 
2.17.1

